cookiecutter/exceptions.py

```diff
@@ -68,3 +68,9 @@ class ContextDecodingException(CookiecutterException):
     """
     Raised when a project's JSON context file can not be decoded.
     """
+
+
+class ParsingError(SyntaxError):
```

Comment:
Don't we need to offer the possibility to make a distinction between standard SyntaxError (which could stem from any python code or other non-controlled places -- like libraries --) and what was identified as ParsingError from one of the parsers of this module (and that can not be generated by any other code as it's unique to this task) ?

The code already rely on this in this very patch to display an error message on what was identified as syntax error *while parsing config file* whatever the format of the config file. It's meant as a generalization of sub-parser's parsing exceptions (namely ``ScannerError`` for Yaml, and ... ``ValueError`` for json). Without this exception, I believe we are at risk of displaying a misleading error message to the user/dev, catching unrelated SyntaxError that could be casted in any context of any inner libraries.